Modules/FIT/FV0/src/OutOfBunchCollFeeModulesCheck.cxx

```diff
@@ -0,0 +1,156 @@
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   OutOfBunchCollFeeModulesCheck.cxx
+/// \author Dawid Skora dawid.mateusz.skora@cern.ch
+///
+
+#include "FV0/OutOfBunchCollFeeModulesCheck.h"
+#include "QualityControl/MonitorObject.h"
+#include "QualityControl/Quality.h"
+#include "QualityControl/QcInfoLogger.h"
+#include "DataFormatsParameters/GRPLHCIFData.h"
+
+// ROOT
+#include <TH1.h>
+#include <TH2.h>
+#include <TPaveText.h>
+#include <TLine.h>
+#include <TList.h>
+
+#include <map>
+#include <DataFormatsQualityControl/FlagReasons.h>
+
+using namespace std;
+using namespace o2::quality_control;
+
+namespace o2::quality_control_modules::fv0
+{
+
+void OutOfBunchCollFeeModulesCheck::configure()
+{
+  if (auto param = mCustomParameters.find("thresholdWarning"); param != mCustomParameters.end()) {
+    mThreshWarning = stof(param->second);
+    ILOG(Debug, Support) << "configure() : using thresholdWarning = " << mThreshWarning << ENDM;
+  } else {
+    mThreshWarning = 1e-3;
+    ILOG(Debug, Support) << "configure() : using default thresholdWarning = " << mThreshWarning << ENDM;
+  }
+
+  if (auto param = mCustomParameters.find("thresholdError"); param != mCustomParameters.end()) {
+    mThreshError = stof(param->second);
+    ILOG(Debug, Support) << "configure() : using thresholdError = " << mThreshError << ENDM;
+  } else {
+    mThreshError = 0.1;
+    ILOG(Debug, Support) << "configure() : using default thresholdError = " << mThreshError << ENDM;
+  }
+
+  if (mThreshError < mThreshWarning) {
+    ILOG(Debug, Support) << "thresholdError lower than thresholdWarning. Swaping values." << ENDM;
+    std::swap(mThreshError, mThreshWarning);
+  }
+}
+
+Quality OutOfBunchCollFeeModulesCheck::check(std::map<std::string, std::shared_ptr<MonitorObject>>* moMap)
+{
+  Quality result = Quality::Null;
+
+  for (auto& [moName, mo] : *moMap) {
+    (void)moName;
+    if (mo->getName() == "OutOfBunchColl_BCvsFeeModules") {
+      auto* histogram = dynamic_cast<TH2*>(mo->getObject());
+
+      if (!histogram) {
+        ILOG(Error, Support) << "check(): MO " << mo->getName() << " not found" << ENDM;
+        result.addReason(FlagReasonFactory::Unknown(), "MO " + mo->getName() + " not found");
+        result.set(Quality::Null);
+        return result;
+      }
+
+      std::vector<float> allCollPerFeeModule(mo->getMetadataMap().size() + 1, 0);
+      for (auto metainfo : mo->getMetadataMap()) {
+        int bin = 0;
+        float value = 0;
+        try {
+          bin = std::stoi(metainfo.first);
+          value = std::stof(metainfo.second);
+        } catch (const std::invalid_argument& e) {
+          continue;
+        }
```

Comment:
I guess exception will break the run. Maybe better to send error message into InfoLogger?

Reply:
Exception shouldn't brake the run, only pass actual 'for' run. However there was a bug related to wrong metadata key. It was fixed by latest commit.